client/login/magic-login/gravatar/index.tsx

```diff
@@ -0,0 +1,1038 @@
+						value={ verificationCodeInputValue }
+						onChange={ handleGravPoweredCodeInputChange }
+						placeholder={ translate( 'Verification code' ) }
+						disabled={ isProcessingCode }
+						isError={ !! codeValidationError }
+						autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+					/>
+					{ codeValidationError && (
+						<Notice
+							text={ errorText }
+							className="magic-login__request-login-email-form-notice"
+							showDismiss={ false }
+							status="is-transparent-info"
+						/>
+					) }
+					<FormButton
+						primary
+						disabled={
+							! verificationCodeInputValue ||
+							verificationCodeInputValue.length < 6 ||
+							isProcessingCode
+						}
+						busy={ isProcessingCode }
+					>
+						{ translate( 'Continue' ) }
+					</FormButton>
+				</form>
+				<footer
+					className={ clsx( 'grav-powered-magic-login__footer', {
+						'grav-powered-magic-login__footer--vertical': ! isFromGravatar3rdPartyApp,
+					} ) }
+				>
+					<button
+						onClick={ () => {
+							handleGravPoweredEmailCodeSend( usernameOrEmail );
+
+							recordTracksEvent( 'calypso_gravatar_powered_magic_login_click_resend_email', {
+								type: 'code',
+								client_id: oauth2ClientId,
+								client_name: oauth2Client?.title,
+							} );
+						} }
+						disabled={ Boolean( isRequestingEmail ) || Boolean( resendEmailCountdown ) }
+					>
+						{ resendEmailCountdown === 0
+							? translate( 'Send again' )
+							: translate( 'Send again (%(countdown)d)', {
+									args: { countdown: resendEmailCountdown },
+							  } ) }
+					</button>
+					{ shouldShowSwitchEmail && (
+						<button
+							onClick={ () => {
+								resetResendEmailCountdown();
+								handleGravPoweredEmailSwitch();
+							} }
+						>
+							{ translate( 'Switch email' ) }
+						</button>
+					) }
+					<a href="https://gravatar.com/support" target="_blank" rel="noreferrer">
+						{ translate( 'Need help logging in?' ) }
+					</a>
+				</footer>
+			</div>
+		);
+	};
+
+	const GravPoweredEmailLinkVerification = () => {
+		const emailAddress =
+			usernameOrEmail && usernameOrEmail.includes( '@' ) ? usernameOrEmail : null;
+		const isSendingEmail = useSelector( isFetchingMagicLoginEmail );
+
+		const emailTextOptions = {
+			components: {
+				sendEmailButton: (
+					<button
+						onClick={ () => {
+							usernameOrEmail &&
+								dispatch(
+									sendEmailLogin( usernameOrEmail, {
+										redirectTo: currentQueryArguments?.redirect_to ?? '', // This is a required field
+										requestLoginEmailFormFlow: true,
+										createAccount: true,
+										flow: oauth2Client ? getGravatarOAuth2Flow( oauth2Client ) ?? '' : '',
+										showGlobalNotices: true,
+										/**
+										 * The below are required fields but not used in this flow
+										 */
+										tokenType: 'link',
+										source: false,
+										blogId: '',
+										loginFormFlow: false,
+										isMobileAppLogin: false,
```

Comment:
@wellyshen these were passed before due to TS not flagging anything. Now they are required.